standard_sequence/experiment_components.py

```diff
@@ -0,0 +1,816 @@
+        )
+        devices.mirror_2_vertical.constant(t, shot_globals.ryd_456_mirror_2_v_position)
+
+    def red_mirror_1_position(self, t):
+        devices.mirror_3_horizontal.constant(
+            t, shot_globals.ryd_1064_mirror_1_h_position
+        )
+        devices.mirror_3_vertical.constant(t, shot_globals.ryd_1064_mirror_1_v_position)
+
+    def red_mirror_2_position(self, t):
+        devices.mirror_4_horizontal.constant(
+            t, shot_globals.ryd_1064_mirror_2_h_position
+        )
+        devices.mirror_4_vertical.constant(t, shot_globals.ryd_1064_mirror_2_v_position)
+
+
+class UVLamps:
+    def __init__(self, t):
+        # Turn off UV lamps
+        devices.uv_switch.go_low(t)
+
+    def uv_pulse(self, t, dur):
+        """Flash the UV LED lamps for dur seconds"""
+        devices.uv_switch.go_high(t)
+        t += dur
+        devices.uv_switch.go_low(t)
+        return t
+
+
+class BField:
+    CONST_COIL_OFF_TIME: ClassVar[float] = (
+        1.4e-3  # minimum time for the MOT coil to be off
+    )
+    CONST_COIL_RAMP_TIME: ClassVar[float] = (
+        100e-6  # ramp time from initial field to final field
+    )
+    CONST_BIPOLAR_COIL_FLIP_TIME: ClassVar[float] = (
+        10e-3  # the time takes to flip the polarity of the coil
+    )
+    CONST_COIL_FEEDBACK_OFF_TIME: ClassVar[float] = (
+        4.5e-3  # how long to turn off the feedback of circuit when flipping polarity
+    )
+
+    bias_voltages = list[float]
+    mot_coils_on: bool
+    mot_coils_on_current: float
+    current_outputs = tuple[AnalogOut, AnalogOut, AnalogOut]
+    feedback_disable_ttls = tuple[DigitalOut, DigitalOut, DigitalOut]
+
+    def __init__(self, t: float):
+        self.bias_voltages = [
+            shot_globals.mot_x_coil_voltage,
+            shot_globals.mot_y_coil_voltage,
+            shot_globals.mot_z_coil_voltage,
+        ]
+        self.mot_coils_on = shot_globals.do_mot_coil
+        self.mot_coils_on_current = 10 / 6
+
+        # Same question as for Dline_lasers, should we automatically initialize the hardware here or in a separate function we can call?
+
+        # initialize bias_field variable to None
+        # self.bias_field = None
+
+        self.t_last_change = 0
+
+        self.current_outputs = (
+            devices.x_coil_current,
+            devices.y_coil_current,
+            devices.z_coil_current,
+        )
+
+        self.feedback_disable_ttls = (
+            devices.x_coil_feedback_off,
+            devices.y_coil_feedback_off,
+            devices.z_coil_feedback_off,
+        )
+
+        for current_output, bias_voltage_cmpnt in zip(
+            self.current_outputs, self.bias_voltages
+        ):
+            current_output.constant(t, bias_voltage_cmpnt)
+
+        if self.mot_coils_on:
+            devices.mot_coil_current_ctrl.constant(t, self.mot_coils_on_current)
+        else:
+            devices.mot_coil_current_ctrl.constant(t, 0)
+
+    def flip_coil_polarity(

```

Comment:
docstring, explain what this does because it is not trivial

Reply:
added